module/VuFind/src/VuFind/Controller/HoldsTrait.php

```diff
@@ -78,6 +78,19 @@ public function holdAction()
         if (!$gatheredDetails) {
             return $this->redirectToRecord();
         }
+
+        // Attach holdings data from requested item for template use
+        $holdings = $catalog->getHolding($driver->getUniqueID(), $patron);
```

Comment:
This is an expensive call; I would try to avoid making it until absolutely necessary. Can/should this logic take place later in the function? If nothing else, shouldn't we skip the holdings lookup if there is no item ID in the request?